src/compiler/tsc.ts

```diff
@@ -360,7 +360,7 @@ namespace ts {
             let newFileNames = ts.map(parsedCommandLine.fileNames, compilerHost.getCanonicalFileName);
             let canonicalRootFileNames = ts.map(rootFileNames, compilerHost.getCanonicalFileName);
 
-            if (!arrayStructurallyIsEqualTo(newFileNames, canonicalRootFileNames)) {
+            if (!arrayIsEqualTo(newFileNames.sort(), canonicalRootFileNames.sort())) {

```

Comment:
Can you add a comment that if the file names are not the same, we just throw away the entire program?